nvflare/fuel/f3/cellnet/cell_cipher.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from cryptography.exceptions import InvalidKey, InvalidSignature
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import padding as sym_padding
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+
+from nvflare.fuel.hci.base64_utils import b64str_to_bytes, bytes_to_b64str
+
+HASH_LENGTH = 4  # Adjustable to avoid collision
+NONCE_LENGTH = 16  # For AES, this is 128 bits (i.e. block size)
+KEY_LENGTH = 32  # AES 256.  Choose from 16, 24, 32
+HEADER_LENGTH = HASH_LENGTH + NONCE_LENGTH
+PADDING_LENGTH = NONCE_LENGTH * 8  # in bits
+
+
+def get_hash(value):
+    hash = hashes.Hash(hashes.SHA256())
+    hash.update(value)
+    return hash.finalize()
+
+
+class SessionKeyUnavailable(Exception):
+    pass
+
+
+class SessionKeyManager:
+    def __init__(self):
```

Comment:
To validate cert, CA certs and intermediate certs are needed. SessionManager constructor needs those information.

Reply:
We only need root CA cert.  There is no intermediate cert.  I will add the root ca cert as argument in the init.